Makefile.menhir

```diff
@@ -157,7 +176,67 @@ include .depend.menhir
 
 interpret-menhir:
 	@ echo "Please wait, I am building the LALR automaton..."
-	@ $(MENHIR) $(MENHIRFLAGS) parsing/parser.mly \
+	@ $(MENHIR) $(MENHIRBASICFLAGS) parsing/parser.mly \
 	    --interpret \
 	    --interpret-show-cst \
 	    --trace \
+
+## list-errors
+
+# This rule runs Menhir's reachability analysis, which produces a list of all
+# states where a syntax error can be detected (and a corresponding list of of
+# erroneous sentences). This data is stored in parsing/parser.auto.messages.
+# This analysis requires about 3 minutes and 6GB of RAM.
+
+# The analysis is performed on a copy of the grammar where every block
+# of text comprised between the markers BEGIN AVOID and END AVOID has
+# been removed. This allows us to avoid certain syntactic forms in the
+# sentences that we produce. See parser.mly for more explanations.
+
+# Because of this, we must run Menhir twice: once on a modified copy of the
+# grammar to produce the sentences, and once on the original grammar to update
+# the auto-comments (which would otherwise be incorrect).
+
+.PHONY: list-errors
+list-errors:
+	@ tmp=`mktemp -d /tmp/parser.XXXX` && \
+	  sed -e '/BEGIN AVOID/,/END AVOID/d' parsing/parser.mly > $$tmp/parser.mly && \
+	  $(MENHIR) $(MENHIRBASICFLAGS) $$tmp/parser.mly \
+	    --list-errors -la 2 \
+	    > parsing/parser.auto.messages && \
+	  rm -rf $$tmp
+	@ cp parsing/parser.auto.messages parsing/parser.auto.messages.bak
+	@ $(MENHIR) $(MENHIRBASICFLAGS) parsing/parser.mly \
+	    --update-errors parsing/parser.auto.messages.bak \
+	    > parsing/parser.auto.messages
+	@ rm -f parsing/parser.auto.messages.bak
+
+## generate-parse-errors
+
+# This rule assumes that [make list-errors] has been run first.
+
+# This rule turns the error sentences stored in parsing/parser.auto.messages
+# into concrete .ml files, which can be used as tests. One file per sentence
+# is created. The file name is derived from the sentence. The test files are
+# placed in the directory testsuite/tests/generated-parse-errors. The previous
+# content of this directory is not erased.
+
+# For speed we use a parallel loop (note & at the end of the loop body). This
+# seems about three times faster than a sequential loop.
```

Comment:
Information on how long this takes on your machine would be helpful to manage user expectations, especially if the command does not include progress output during its computation.

Reply:
OK, information added (5s versus 15s).